### PR TITLE
ab-888: adding semgrep fatal_errors message

### PR DESCRIPTION
**Pull request by Anthony-Mckale**
- added fatal_errors to scan results object
- change exception to fatal_error object for semgrep object
- updated console reporter to display results as table so looks better
- updated console reporter to display results with errors

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -102,8 +103,12 @@ RUN pip3 install --no-cache-dir cyclonedx-bom && echo "SIZETAG:Tool:python/cyclo
 
 #
 ## Install Anchore tools
-RUN curl -sSfL https://raw.githubusercontent.com/anchore/grype/main/install.sh | sh -s -- -b /usr/local/bin  && echo "SIZETAG:Tool:anchore/grype"
-RUN curl -sSfL https://raw.githubusercontent.com/anchore/syft/main/install.sh | sh -s -- -b /usr/local/bin  && echo "SIZETAG:Tool:anchore/syft"
+RUN set -o pipefail
```
Comment (mgonzales-rs) [inline Dockerfile:106]: I think we need a backslash to write multiple lines.
Otherwise there will be an error creating the Docker image
image

**Review by mgonzales-rs — CHANGES_REQUESTED** [review]
Some commands are failing:

```
$ eze tools list
Traceback (most recent call last):
  File "C:\Users\Riversafe\.pyenv\pyenv-win\versions\3.8.10\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Riversafe\.pyenv\pyenv-win\versions\3.8.10\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\dev\repos\eze-cli\.venv\Scripts\eze.exe\__main__.py", line 4, in <module>
  File "c:\dev\repos\eze-cli\.venv\lib\site-packages\eze\cli\main.py", line 16, in <module>
    from eze.utils.package import get_plugins
  File "c:\dev\repos\eze-cli\.venv\lib\site-packages\eze\utils\package.py", line 30, in <module>
    from eze.plugins import base_plugins
  File "c:\dev\repos\eze-cli\.venv\lib\site-packages\eze\plugins\base_plugins.py", line 8, in <module>
    from eze.plugins.reporters.console import ConsoleReporter
  File "c:\dev\repos\eze-cli\.venv\lib\site-packages\eze\plugins\reporters\console.py", line 18, in <module>
    class ConsoleReporter(ReporterMeta):
  File "c:\dev\repos\eze-cli\.venv\lib\site-packages\eze\plugins\reporters\console.py", line 73, in ConsoleReporter
    def print_scan_summary_table(self, scan_results: list[ScanResult]):
TypeError: 'type' object is not subscriptable
```

```
$ eze test
Traceback (most recent call last):
  File "C:\Users\Riversafe\.pyenv\pyenv-win\versions\3.8.10\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Riversafe\.pyenv\pyenv-win\versions\3.8.10\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\dev\repos\eze-cli\.venv\Scripts\eze.exe\__main__.py", line 4, in <module>
  File "c:\dev\repos\eze-cli\.venv\lib\site-packages\eze\cli\main.py", line 16, in <module>
    from eze.utils.package import get_plugins
  File "c:\dev\repos\eze-cli\.venv\lib\site-packages\eze\utils\package.py", line 30, in <module>
    from eze.plugins import base_plugins
  File "c:\dev\repos\eze-cli\.venv\lib\site-packages\eze\plugins\base_plugins.py", line 8, in <module>
    from eze.plugins.reporters.console import ConsoleReporter
  File "c:\dev\repos\eze-cli\.venv\lib\site-packages\eze\plugins\reporters\console.py", line 18, in <module>
    class ConsoleReporter(ReporterMeta):
  File "c:\dev\repos\eze-cli\.venv\lib\site-packages\eze\plugins\reporters\console.py", line 73, in ConsoleReporter
    def print_scan_summary_table(self, scan_results: list[ScanResult]):
TypeError: 'type' object is not subscriptable
```

File: eze/plugins/reporters/console.py [hunk 2]
```diff
@@ -49,34 +49,84 @@ async def run_report(self, scan_results: list):
...
-            return
-
+        self._print_scan_report_errors(scan_results_with_errors)
+        self._print_scan_report_warnings(scan_results_with_warnings)
         self._print_scan_report_vulnerabilities(scan_results_with_vulnerabilities)
         self._print_scan_report_sbom(scan_results_with_sboms)
-        self._print_scan_report_warnings(scan_results_with_warnings)
 
-    def print_scan_summary_title(self, scan_result: ScanResult) -> str:
+    def print_scan_summary_table(self, scan_results: list[ScanResult]):
```
Comment (mgonzales-rs) [inline eze/plugins/reporters/console.py:73]: All errors refer this line.

Comment (Anthony-Mckale) [reply eze/plugins/reporters/console.py:73]: i was on 3.9 ... generics must be a 3.9 feature, removing


```
Riversafe@DESKTOP-GPVF0RR MINGW64 /c/dev/repos/eze-cli (feature/ab-850-node-cyclonedx-goat)
$ python --version
Python 3.9.7
```

Comment (Anthony-Mckale) [reply eze/plugins/reporters/console.py:73]: we'll see if the code works now

File: Dockerfile (hunk 1 above)
Comment (Anthony-Mckale) [reply Dockerfile:106]: so this works for me, think you might of tested it, while i was still WIP

**Review by mgonzales-rs — APPROVED** [review]

**Pull request merged**